libgstd/libgstd.c

```diff
@@ -0,0 +1,198 @@
+/*
+ * GStreamer Daemon - gst-launch on steroids
+ * C library abstracting gstd
+ *
+ * Copyright (c) 2015-2021 RidgeRun, LLC (http://www.ridgerun.com)
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#define _GNU_SOURCE
+#include <stdarg.h>
+#include <stdio.h>
+
+#include "libgstd.h"
+#include "gstd_ipc.h"
+#include "gstd_tcp.h"
+#include "gstd_unix.h"
+#include "gstd_http.h"
+#include "gstd_log.h"
+#include "libgstd_assert.h"
+
+static GType gstd_supported_ipc_to_ipc (SupportedIpcs code);
+static void gstd_manager_init (GOptionGroup ** gst_group, int argc,
+    char *argv[]);
+
+struct _GstDManager
+{
+  GstdSession *session;
+  GstdIpc **ipc_array;
+  guint num_ipcs;
+};
+
+static GType
+gstd_supported_ipc_to_ipc (SupportedIpcs code)
+{
+  GType code_description[] = {
+    [GSTD_IPC_TYPE_TCP] = GSTD_TYPE_TCP,
+    [GSTD_IPC_TYPE_UNIX] = GSTD_TYPE_UNIX,
+    [GSTD_IPC_TYPE_HTTP] = GSTD_TYPE_HTTP
+  };
+
+  const gint size = sizeof (code_description) / sizeof (gchar *);
+
+  gstd_assert_and_ret_val (0 <= code, GSTD_TYPE_IPC);   // TODO: Proponer un GSTD_TYPE_DEFAULT
```

Comment:
Don't leave `TODO`s in the code, we should never leave dirty code committed. If you need to address this later create an issue or a task